packages/preferences/src/browser/preferences-contribution.ts

```diff
@@ -69,22 +95,29 @@ export class PreferencesContribution extends AbstractViewContribution<Preference
                 keybinding: 'cmd+,'
             });
         }
+
         keybindings.registerKeybinding({
             command: CommonCommands.OPEN_PREFERENCES.id,
-            keybinding: 'ctrl+,'
+            keybinding: 'ctrl+,',
         });
     }
 
-    protected async openPreferences(preferenceScope: PreferenceScope): Promise<void> {
-        const wsUri = this.workspacePreferenceProvider.getConfigUri();
-        if (wsUri && !await this.filesystem.exists(wsUri.toString())) {
-            await this.filesystem.createFile(wsUri.toString());
-        }
-
-        const widget = await this.widget;
-        widget.preferenceScope = preferenceScope;
-        super.openView({ activate: true });
-        widget.activatePreferenceEditor(preferenceScope);
+    registerToolbarItems(toolbar: TabBarToolbarRegistry): void {
+        toolbar.registerItem({
+            id: PreferencesCommands.OPEN_PREFERENCES_JSON_TOOLBAR.id,
+            command: PreferencesCommands.OPEN_PREFERENCES_JSON_TOOLBAR.id,
+            tooltip: 'Open Preferences in JSON',
+            priority: 0,
+        });
     }
 
+    /**
+     * Determine if the current widget is the keybindings widget.
```

Comment:
[minor] - determine if the current widget is the **preferences** widget - not the *keybindings**